test/astar.cc

```diff
@@ -962,6 +962,92 @@ void test_time_restricted_road() {
   }
 }
 
+Api route_on_timerestricted(std::string& costing_str, int16_t hour) {
+  // Try routing over "Via Montebello" in Rome which is a time restricted road
+  // The restriction is
+  //
+  //    <tag k="auto" v="yes"/>
+  //    <tag k="motor_vehicle:conditional" v="no @ (Mo-Sa 07:00-16:00)"/>
+  //    <tag k="pedestrian" v="no"/>
+  //    <tag k="pedestrian:conditional" v="yes @ (Mo-Sa 08:00-15:00)"/>
+  //
+  // so lets use a timedependent a-star and verify that
+
+  LOG_INFO("Testing " + costing_str + " route at hour " + std::to_string(hour));
+  auto conf = get_conf("roma_tiles");
+  route_tester tester(conf);
+  // The following request results in timedep astar during the restricted hours
+  // and should be denied
+  std::string request =
+      R"({
+        "locations":[{"lat":41.90550,"lon":12.50090},{"lat":41.90477,"lon":12.49914}],
+        "costing":")" +
+      costing_str + R"(",
+          "date_time":{
+            "type":1,
+            "value":"2020-01-16T)" +
+      std::to_string(hour) + R"(:05"
+          }
+        })";
+
+  return tester.test(request);
+}
+
+void test_route_restricted(std::string costing_str, int16_t hour) {
+  bool found_route = false;
+  try {
+    auto response = route_on_timerestricted(costing_str, hour);
+    found_route = true;
+    const auto& leg = response.directions().routes(0).legs(0);
+    LOG_INFO("Route that wasn't supposed to happen: " + leg.shape());
+  } catch (const std::exception& e) {
+    if (std::string(e.what()) != "No path could be found for input") {
+      throw std::logic_error("Was expecting 'No path could be found for input'");
+    } else {
+      return;
+    }
+  }
+  if (found_route) {
+    throw std::logic_error("Found a route when no route was expected");
+  }
+}
+
+void test_time_restricted_road_denied_on_timedep() {
+  {
+    // A car at hour 11 should be denied
+    std::string costing_str("auto");
+    test_route_restricted(costing_str, 11);
+  }
+  {
+    // A pedestrian at hour 22 should be denied
+    std::string costing_str("pedestrian");
+    test_route_restricted(costing_str, 22);
+  }
+}
+
+void test_route_allowed(std::string costing_str, int16_t hour) {
+  auto response = route_on_timerestricted(costing_str, hour);
+  const auto& legs = response.trip().routes(0).legs();
```

Comment:
Looks like legs is unused here. Did you plan to add an assertion here like legs.size() == 1 ?

Reply:
Good catch! :+1:

Reply:
Pushed new code to test for legs.size() == 1

Reply:
> ning for unused variable?

We're not yet treating all warnings as errors unfortunately.